### ISSUE TITLE
minetest.get_all_craft_recipes doesn't return "fuel" recipes

### ISSUE THREAD

**Issue by FaceDeer** [luanti-org/luanti#5745]
I've found that get_all_craft_recipes will return "normal" recipes and "cooking" recipes for an item, but it doesn't return "fuel" recipes. I've had to work around this by calling `minetest.get_craft_result({method="fuel",width=1,items={item}})` on every item and checking to see if it gives a result.

#4901 may be a semi-related issue.

**Comment by nerzhul** [comment]
get_all_craft_recipes look at m_output_craft_definitions fuel has no output then it seems logical.

@paramat @sofar have you an opinion about this ?

**Comment by paramat** [comment]
Fuel 'recipes' are not really recipes even though they use that word. So i feel more of a request than a bug, and not priority for release.

**Comment by paramat** [comment]
1+ year bump, any core dev support?

**Comment by Wuzzy2** [comment]
How is this not considered a bug?
It's frustrating to learn as modder it is not possible to get fuel recipes like that. It's inconsistent.

Whether fuel recipes are considered “real” recipes is besides the point: They are considered recipes internally and in the Lua API, and only that should count.

**Comment by rubenwardy** [comment]
I support this

**Comment by paramat** [comment]
Fuel doesn't seem a 'craft recipe' to me, and there's no craft result, even though codewise it uses 'minetest.register_craft()' so it's debateable. It seems a strange way to register a fuel.
So as a modder i would not expect fuels to be returned by 'get craft recipes'. But i don't mind if it does.

**Comment by pauloue** [comment]
I don't think it would make sense for `get_all_craft_recipes` (or `get_craft_recipe`) to return fuel recipes. How would you call it without an output? It would have to use a special output like "burn" (and it still wouldn't be easy to check if some item is a fuel or see what it's burntime is).

Maybe add fuel recipes to a table (`core.registered_fuels`) instead? Then it would be easy to both get all fuels and check if some item is a fuel.

**Comment by jastevenson303** [comment]
both fuels and cooking use register_craft, maybe others.

**Comment by jastevenson303** [comment]
How about an option to `get_all_craft_recipes` that allows to specify type?

**Comment by Skamiz** [comment]
Bump. Just run into this in the wild and want to note, as of 5.8.0 you CAN use `minetest.get_all_craft_recipes` to find out fuel recipes. Specifically like this: `minetest.get_all_craft_recipes("")` by querying for an empty string.

The returned recipes look like this:
```
{
	width = 40,
	output = "",
	items = {
		"default:coal_lump"
	},
	method = "cooking",
	type = "cooking"
},
```
empty output and and 'width' corresponding to 'cooktime'. Note that the method is listed as cooking, instead of fuel.

I don't know in what form recipes are stored in the engine, but half the time they are requested back, they are returned in a mangled form and it's up to the modder to make sense of them again. See also how shapeless recipes are signified by a width of 0, and replacements are completely left out, requiring tedious workarounds to be found out #4901
